src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/net/GRPCConnectionManager.java

```diff
@@ -43,24 +47,37 @@
   private static final Logger LOG = LogManager.getLogger(GRPCConnectionManager.class);
   private static final String EMPTY_STRING = "";
 
-  private Map<String, ManagedChannel> perHostChannelMap = new HashMap<>();
-  private Map<String, InterNodeRpcServiceGrpc.InterNodeRpcServiceStub> perHostClientStubMap =
-      new HashMap<>();
+  private ConcurrentMap<String, AtomicReference<ManagedChannel>> perHostChannelMap =
+      new ConcurrentHashMap<>();
+  private ConcurrentMap<String, AtomicReference<InterNodeRpcServiceStub>> perHostClientStubMap =
+      new ConcurrentHashMap<>();
 
   private final boolean shouldUseHttps;
 
   public GRPCConnectionManager(final boolean shouldUseHttps) {
     this.shouldUseHttps = shouldUseHttps;
   }
 
-  public InterNodeRpcServiceGrpc.InterNodeRpcServiceStub getClientStubForHost(
+  public InterNodeRpcServiceStub getClientStubForHost(
```

Comment:
Let's add a line of comment for public method

Reply:
Added.